unit_tests/test_wait.py

```diff
@@ -79,6 +81,109 @@ def callback(arg1, arg2):
         self.assertEqual(wait_for(callback, timeout=2, step=0.5, arg1=1, arg2=3, throw_exc=False), 'what ever')
         self.assertEqual(len(calls), 1)
 
+    def test_04_stop_by_event(self):
+        calls = []
+
+        def callback(arg1, arg2):
+            calls.append((arg1, arg2))
+            if len(calls) == 5:
+                return "what ever"
+            return False
```

Comment:
This is defined separately for each of the unit tests. Please, remove the code duplication.

Reply:
fixed